libbeat/tests/integration/framework.go

```diff
@@ -72,27 +77,80 @@ func NewBeat(t *testing.T, beatName, binary string, args ...string) BeatProc {
 }
 
 // Start starts the Beat process
-// args are extra arguments to be passed to the Beat
+// args are extra arguments to be passed to the Beat.
 func (b *BeatProc) Start(args ...string) {
 	t := b.t
-	b.Args = append(b.Args, args...)
+
 	fullPath, err := filepath.Abs(b.Binary)
 	if err != nil {
 		t.Fatalf("could not get full path from %q, err: %s", b.Binary, err)
 	}
-	b.Cmd = exec.Command(fullPath, b.Args...)
 
-	if err := b.Cmd.Start(); err != nil {
-		t.Fatalf("could not start process: %s", err)
+	b.fullPath = fullPath
+	b.Args = append(b.Args, args...)
+
+	done := atomic.MakeBool(false)
+	wg := sync.WaitGroup{}
+	if b.RestartOnBeatOnExit {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for !done.Load() {
+				b.startBeat()
+				b.waitBeatToExit()
+			}
+		}()
+	} else {
+		b.startBeat()
 	}
+
 	t.Cleanup(func() {
-		pid := b.Cmd.Process.Pid
+		b.cmdMutex.Lock()
+		// 1. Kill the Beat
```

Comment:
Why not use `b.Cmd.Process.Kill` instead of `Signal(os.Interrupt)`. SIGINT isn't guaranteed to cause the Beat to actually exit.

Reply:
Also you don't call `waitForBeatToExit` you aren't guaranteed that the process is stopped when this function returns. It could still exit, but sometime into the execution of a subsequent test.

Reply:
`os.Kill` will force quit the Beat, skipping many of the shutdown process, I wanted to avoid this. The binary we use  is the "system test binary" that effectively is a Beat wrapped on a Go test, so signal handling is a bit different there.

Also we listen to `os.Interrupt`: https://github.com/elastic/beats/blob/e3367eddcb6f03c3d7c38e9735999be1a9ca1cb9/x-pack/libbeat/management/managerV2.go#L473